packages/sync/src/replicastore/class-table-checksum.php

```diff
@@ -0,0 +1,524 @@
+	 *
+	 * @var string
+	 */
+	public $additional_filter_sql = '';
+
+	/**
+	 * Default Checksum Table Configurations.
+	 *
+	 * @var array
+	 */
+	public $default_tables = array();
+
+	/**
+	 * Salt to be used when generating checksum.
+	 *
+	 * @var string
+	 */
+	public $salt = '';
+
+	/**
+	 * Tables which are allowed to be checksummed.
+	 *
+	 * @var string
+	 */
+	public $allowed_tables = array();
+
+	/**
+	 * Table_Checksum constructor.
+	 *
+	 * @param string $table
+	 * @param string $salt
+	 *
+	 * @throws Exception
+	 */
+	public function __construct( $table, $salt = null ) {
+		$this->salt = $salt;
+
+		$this->default_tables = $this->get_default_tables();
+
+		// TODO change filters to allow the array format.
+		// TODO add get_fields or similar method to get things out of the table.
+		// TODO extract this configuration in a better way, still make it work with `$wpdb` names.
+		// TODO take over the replicastore functions and move them over to this class.
+		// TODO make the API work.
+
+		$this->allowed_tables = apply_filters( 'jetpack_sync_checksum_allowed_tables', $this->default_tables );
+
+		$this->table               = $this->validate_table_name( $table );
+		$this->table_configuration = $this->allowed_tables[ $table ];
+
+		$this->prepare_fields( $this->table_configuration );
+
+	}
+
+	/**
+	 * Get Default Table configurations.
+	 *
+	 * @return array
+	 */
+	private function get_default_tables() {
+		global $wpdb;
+
+		 return array(
+			 'posts'              => array(
+				 'table'           => $wpdb->posts,
+				 'range_field'     => 'ID',
+				 'key_fields'      => array( 'ID' ),
+				 'checksum_fields' => array( 'post_modified_gmt' ),
+				 'filter_sql'      => Settings::get_blacklisted_post_types_sql(),
+			 ),
+			 'postmeta'           => array(
+				 'table'           => $wpdb->postmeta,
+				 'range_field'     => 'post_id',
+				 'key_fields'      => array( 'post_id', 'meta_key' ),
+				 'checksum_fields' => array( 'meta_key', 'meta_value' ),
+				 'filter_sql'      => Settings::get_whitelisted_post_meta_sql(),
+			 ),
+			 'comments'           => array(
+				 'table'           => $wpdb->comments,
+				 'range_field'     => 'comment_ID',
+				 'key_fields'      => array( 'comment_ID' ),
+				 'checksum_fields' => array( 'comment_content' ),
+				 'filter_sql'      => Settings::get_comments_filter_sql(),
+			 ),
+			 'commentmeta'        => array(
+				 'table'           => $wpdb->commentmeta,
+				 'range_field'     => 'comment_id',
+				 'key_fields'      => array( 'comment_id', 'meta_key' ),
+				 'checksum_fields' => array( 'meta_key', 'meta_value' ),
+				 'filter_sql'      => Settings::get_whitelisted_comment_meta_sql(),
+			 ),
+			 'terms'              => array(
+				 'table'           => $wpdb->terms,
+				 'range_field'     => 'term_id',
+				 'key_fields'      => array( 'term_id' ),
+				 'checksum_fields' => array( 'term_id', 'name', 'slug' ),
+			 ),
+			 'termmeta'           => array(
```

Comment:
We don't actively sync term meta. This is fine to leave here but should not be included in the checksum all.

When we do include it it needs to filter everything from terms (get_blacklisted_taxonomies_sql)